src/logic/updateSessionStorage.js

```diff
@@ -1,7 +1,19 @@
 // @flow
+import logError from '../utils/logError';
 
 export default function updateSessionStorage(src: string) {
-  const cachedImages = JSON.parse(window.sessionStorage.getItem('__REACT_SIMPLE_IMG__')) || {};
+  let cachedImagesRefString = null;
+  try {
+    cachedImagesRefString = window.sessionStorage.getItem('__REACT_SIMPLE_IMG__');
+  } catch (e) {
+    logError(`sessionStorage is disabled ${e}`);
+    return;
+  }
+  const cachedImages = JSON.parse(cachedImagesRefString) || {};
   cachedImages[src] = +new Date();
-  window.sessionStorage.setItem('__REACT_SIMPLE_IMG__', JSON.stringify(cachedImages));
+  try {
+    window.sessionStorage.setItem('__REACT_SIMPLE_IMG__', JSON.stringify(cachedImages));
```

Comment:
it's likely that this won't fail if we get past the above try-catch and the return on line 10, but including for extra safety

Reply:
any chance we can combine the two `try` and `catch` into one?

Reply:
done!